modules/landing-pages/module.php

```diff
@@ -212,53 +269,169 @@ private function register_landing_page_cpt() {
 			'show_in_menu' => 'edit.php?post_type=elementor_library&tabs_group=library',
 			'capability_type' => 'page',
 			'taxonomies' => [ Source_Local::TAXONOMY_TYPE_SLUG ],
+			'hierarchical' => true,
 			'supports' => [ 'title', 'editor', 'comments', 'revisions', 'trackbacks', 'author', 'excerpt', 'page-attributes', 'thumbnail', 'custom-fields', 'post-formats', 'elementor' ],
 		];
 
-		if ( false !== strpos( $this->permalink_structure, '/%postname%/' ) ) {
-			// For rewriting the Landing Page's permalink to act like a page in case the site's permalink structure
-			// is '/%postname%/'.
-			$args['rewrite'] = [
-				'slug' => '/',
-				'with_front' => false,
-			];
-		}
-
 		register_post_type( self::CPT, $args );
 	}
 
-	public function remove_post_type_slug( $post_link, $post, $leavename ) {
+	/**
+	 * Remove Post Type Slug
+	 *
+	 * Landing Pages are supposed to act exactly like pages. This includes their URLs being directly under the site's
+	 * domain name. Since "Landing Pages" is a CPT, WordPress automatically adds the landing page slug as a prefix to
+	 * it's posts' permalinks. This method checks if the post's post type is Landing Pages, and if it is, it removes
+	 * the CPT slug from the requested post URL.
+	 *
+	 * Runs on the 'post_type_link' filter.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @param $post_link
+	 * @param $post
+	 * @param $leavename
+	 * @return string|string[]
+	 */
+	private function remove_post_type_slug( $post_link, $post, $leavename ) {
+		// Only try to modify the permalink if the post is a Landing Page.
 		if ( self::CPT !== $post->post_type || 'publish' !== $post->post_status ) {
 			return $post_link;
 		}
 
-		return str_replace( '/' . $post->post_type . '/', '/', $post_link );
+		// Any slug prefixes need to be removed from the post link.
+		return get_site_url() . '/' . $post->post_name;
```

Comment:
```suggestion
		return get_home_url() . '/' . $post->post_name;
```

Reply:
Why? @matipojo

Reply:
The site URL may be different from home.

Reply:
Done.